pkg/delete/delete.go

```diff
@@ -46,37 +46,56 @@ type DeleteEntries = map[configurationType][]pointer.DeletePointer
+	var deleteErrors int
+
+	// Delete automation resources (in the specified order
+	automationTypeOrder := []string{"workflow", "scheduling-rule", "business-calendar"}
+	for _, key := range automationTypeOrder {
+		entries := entriesToDelete[key]
+		if reflect.ValueOf(clients.Automation).IsNil() {
+			log.WithCtxFields(ctx).WithFields(field.Type(key)).Warn("Skipped deletion of %d Automation configuration(s) of type %q as API client was unavailable.", len(entries), key)
+			delete(entriesToDelete, key)
 			continue
 		}
+		err := automation.Delete(ctx, clients.Automation, automationResources[key], entries)
+		if err != nil {
+			log.WithFields(field.Error(err)).Error("Error during deletion: %v", err)
+			deleteErrors += 1
+		}
+		delete(entriesToDelete, key)
+	}
 
-		var err error
+	// Delete bucket resources
+	if reflect.ValueOf(clients.Buckets).IsNil() {
+		log.WithCtxFields(ctx).WithFields(field.Type("bucket")).Warn("Skipped deletion of %d Grail Bucket configuration(s) as API client was unavailable.", len(entriesToDelete["bucket"]))
+	} else {
+		if err := bucket.Delete(ctx, clients.Buckets, entriesToDelete["bucket"]); err != nil {
+			log.WithFields(field.Error(err)).Error("Error during deletion: %v", err)
+			deleteErrors += 1
+		}
+		delete(entriesToDelete, "bucket")
+	}
 
+	// Dashboard share settings cannot be deleted
+	if _, ok := entriesToDelete[api.DashboardShareSettings]; ok {
+		log.Warn("Classic config of type %s cannot be deleted. Note, that they can be removed by deleting the associated dashboard.", api.DashboardShareSettings)
+		delete(entriesToDelete, api.DashboardShareSettings)
+
+	}
+
+	// Delete classic configs and settings
+	for entryType, entries := range entriesToDelete {
+		var err error
 		if theAPI, isClassicAPI := apis[entryType]; isClassicAPI {
 			err = classic.Delete(ctx, clients.Classic, theAPI, entries)
-		} else if targetAutomation, isAutomationAPI := automationResources[entryType]; isAutomationAPI {
-			if reflect.ValueOf(clients.Automation).IsNil() {
-				log.WithCtxFields(ctx).WithFields(field.Type(entryType)).Warn("Skipped deletion of %d Automation configuration(s) of type %q as API client was unavailable.", len(entries), entryType)
-				continue
-			}
-			err = automation.Delete(ctx, clients.Automation, targetAutomation, entries)
-		} else if entryType == "bucket" {
-			if reflect.ValueOf(clients.Buckets).IsNil() {
-				log.WithCtxFields(ctx).WithFields(field.Type(entryType)).Warn("Skipped deletion of %d Grail Bucket configuration(s) as API client was unavailable.", len(entries))
-				continue
-			}
-			err = bucket.Delete(ctx, clients.Buckets, entries)
 		} else { // assume it's a Settings Schema
 			err = setting.Delete(ctx, clients.Settings, entries)
 		}
-
 		if err != nil {
 			log.WithFields(field.Error(err)).Error("Error during deletion: %v", err)
 			deleteErrors += 1
 		}
+		delete(entriesToDelete, entryType)
```

Comment:
is there any need to delete the entries here? If we don't have to, why modify the thing we're looping over already

Reply:
not really, knowing that the map is empty at the end just has a calming effect :D